### PR TITLE
Add the `sv_url()` rule function

### PR DESCRIPTION
**Pull request by rich-iannone**
This adds the `sv_url()` function and several tests to ensure that URL validation works well.

### CODE REVIEW

File: tests/testthat/test-rules.R [hunk 1]
```diff
@@ -545,6 +545,93 @@ test_that("the `sv_email()` rule function works properly", {
...
 })
 
+test_that("the `sv_url()` rule function works properly", {
+  
+  always_pass <-
+    c(
+      "http://foo.com/blah_blah",
+      "http://foo.com/blah_blah/",
+      "http://foo.com/blah_blah_(wikipedia)",
+      "http://foo.com/blah_blah_(wikipedia)_(again)",
```
Comment (jcheng5) [inline tests/testthat/test-rules.R:555]:
```suggestion
      "http://\u2605foo.com/blah_blah_(wikipedia)_(again)",
```

File: R/rules.R [hunk 2]
```diff
@@ -238,6 +238,73 @@ sv_email <- function(message = "Not a valid email address",
...
+      allow_na = allow_na,
+      allow_nan = FALSE,
+      allow_inf = FALSE
+    ),
+    function(value) {
+      
+      # Regular expression taken from
+      # https://gist.github.com/dperini/729294
+      res <-
+        vapply(
```
Comment (jcheng5) [inline R/rules.R:285]: grepl is vectorized, no need for vapply.

File: tests/testthat/test-rules.R [hunk 3]
```diff
@@ -545,6 +545,93 @@ test_that("the `sv_email()` rule function works properly", {
...
+      "http://0.0.0.0",
+      "http://3628126748",
+      "http://.www.foo.bar/",
+      "http://www.foo.bar./",
+      "http://.www.foo.bar./"
+    )
+  
+  pass_if_multiple <- list(c("http://foo.com/blah_blah", "http://foo.com/blah_blah/"))
+  pass_if_na <- list(NA_integer_)
+  pass_if_multiple_na <- list(c("http://foo.com/blah_blah", NA))
```
Comment (jcheng5) [inline tests/testthat/test-rules.R:616]: Make sure that all three of these lists are represented in each rule's expectations below.

Comment (rich-iannone) [reply tests/testthat/test-rules.R:616]: Now fixed, and was a incomplete in lots of other places so I ensured all possible combinations were satisfied throughout.

File: R/rules.R (hunk 2 above)
Comment (rich-iannone) [reply R/rules.R]: Fixed.

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**